raft/src/main/java/org/apache/kafka/raft/QuorumState.java

```diff
@@ -481,6 +480,7 @@ public void prospectiveAddVotedState(
         int epoch,
         ReplicaKey candidateKey
     ) {
+        prospectiveStateOrThrow();
```

Comment:
Write a comment as to why you are doing this. To me this is not obvious at first glance.